src/content/guides/packages.md

```diff
@@ -261,6 +261,184 @@ To update `pubspec.lock` run `dart pub get` without `--enforce-lockfile`.
+  separate analysis contexts for each package (increasing memory usage).
+
+Pub allows you to organize your repository as a "workspace" using a single
+shared resolution for all your packages.
+
+:::note
+Using a single shared dependency resolution for all your packages increases
+the risks of dependency conflicts. Because Dart does not allow multiple versions
+of the same package.
+
+If the packages are going to be used together (as is commonly the case),
+this is often a feature, as it forces you to resolve incompatibilities between your
+packages when they arise, rather then when you start using the packages.
+:::
+
+To create a workspace:
+
+* Add a `pubspec.yaml` at the repository root directory with a `workspace` entry
+  enummerating the paths to the packages of the respository (the workspace
+  packages):
+
+```yaml
+name: _
+publish_to: none
+environment:
+  sdk: ^3.6.0
+workspace:
+  - packages/helper
+  - packages/client_package
+  - packages/server_package
+```
+
+* For each of the existing `pubspec.yaml`s, make sure their SDK constraint is at
+  least `^3.6.0` and add a `resolution` entry:
+
+  ```yaml
+  environment:
+    sdk: ^3.6.0
+  resolution: workspace
+  ```
+
+* Run `dart pub get` anywhere in the repository, and this will:
+  * Create a single `pubspec.lock` next to the root `pubspec.yaml` that contains
+    the a resolution of all the `dependencies` and `dev_dependencies` of all the
+    workspace packages. 
+  * Create a single shared `.dart_tool/package_config.json` that maps package
+    names to file locations.
+  * Delete any other existing `pubspec.lock` and
+    `.dart_tool/package_config.json` files next to workspace packages.
+
+Now the file structure looks like this:
+
+```console
+/
+  packages/
+    shared/
+      pubspec.yaml
+    client_package/
+      pubspec.yaml
+    server_package/
+      pubspec.yaml
+  pubspec.yaml
+  pubspec.lock
+  .dart_tool/package_config.json
+```
+
+:::version-note
+Pub workspaces was introduced in Dart 3.6.0.
+
+To use pub workspaces, all your workspace packages (but not your dependencies)
+must have a SDK version constraint of `^3.6.0` or higher.
+:::
+
+### Interdependencies between workspace packages
+
+If any of the workspace packages depend on each other, they will automatically
+resolve to the one in the workspace, regardless of the source.
+
+Eg. `packages/client_package/pubspec.yaml` might depend on `shared`:
+
+```yaml
+dependencies:
+  shared: ^2.3.0
+```
+
+When resolved inside the workspace, the _local_ version of `shared` will be
+used. 
+
+The local version of `shared` would still have to match the constraint
+(`^2.3.0`) though.
+
+But when the package is consumed as a dependency without being part of the
+workspace, the original source (here implicitly `hosted`) is used.
+
+So if `client_package` is published to pub.dev and someone depends on it, they
+will get the hosted version of `shared` as a transitive dependency.
```

Comment:
Should this mention of "hosted" also be code-fonted, like the previous paragraph's mention? Or are you just saying that it _is_ hosted, not that that's a package name or anything like that?

```suggestion
will get the `hosted` version of `shared` as a transitive dependency.
```